saleor/product/filters.py

```diff
@@ -26,28 +27,30 @@ class Meta:
         filter_overrides = {PriceField: {'filter_class': RangeFilter}}
 
     def __init__(self, *args, **kwargs):
-        self.category = kwargs.pop('category')
         super().__init__(*args, **kwargs)
         self.product_attributes, self.variant_attributes = (
             self._get_attributes())
         self.filters.update(self._get_product_attributes_filters())
         self.filters.update(self._get_product_variants_attributes_filters())
         self.filters = OrderedDict(sorted(self.filters.items()))
-        self.form.fields['sort_by'].validators.append(self.validate_sort_by)
 
     def _get_attributes(self):
+        q = self._get_attributes_lookup()
         product_attributes = (
             ProductAttribute.objects.all()
             .prefetch_related('values')
-            .filter(product_types__products__category=self.category)
+            .filter(q[0])
```

Comment:
Maybe something like to separate methods for each query would be more readable? Like `_get_product_attributes_lookup` and `_get_variant_attributes_lookup`?